spyder/plugins/toolbar/container.py

```diff
@@ -0,0 +1,322 @@
+            mainwindow.addToolBar(toolbar)
+
+    def add_item_to_application_toolbar(self, item, toolbar, toolbar_id=None,
+                                        section=None, before=None):
+        """
+        Add action or widget `item` to given application menu `section`.
+
+        Parameters
+        ----------
+        item: SpyderAction or QWidget
+            The item to add to the `toolbar`.
+        toolbar: ApplicationToolBar or None
+            Instance of a Spyder application toolbar.
+        toolbar_id: str or None
+            The application toolbar unique string identifier.
+        section: str or None
+            The section id in which to insert the `item` on the `toolbar`.
+        before: str or None
+            Make the item appear before another given item.
+
+        Notes
+        -----
+        Must provide a `toolbar` or a `toolbar_id`.
+        """
+        if toolbar and toolbar_id:
+            raise SpyderAPIError('Must provide only toolbar or toolbar_id!')
+
+        if toolbar is None and toolbar_id is None:
+            raise SpyderAPIError(
+                'Must provide at least toolbar or toolbar_id!')
+
+        if toolbar and not isinstance(toolbar, ApplicationToolBar):
+            raise SpyderAPIError('Not an `ApplicationToolBar`!')
+
+        if toolbar_id and toolbar_id not in self._APPLICATION_TOOLBARS:
+            raise SpyderAPIError(
+                '{} is not a valid toolbar_id'.format(toolbar_id))
+
+        toolbar_id = toolbar_id if toolbar_id else toolbar.ID
+        toolbar = toolbar if toolbar else self.get_application_toolbar(
+            toolbar_id)
+
+        toolbar.add_item(item, section=section, before=before)
+
+    def get_application_toolbar(self, toolbar_id):
+        """
+        Return an application toolbar by toolbar_id.
+
+        Parameters
+        ----------
+        toolbar_id: str
+            The toolbar unique string identifier.
+
+        Returns
+        -------
+        spyder.api.widgets.toolbars.ApplicationToolBar
+            The application toolbar.
+        """
+        if toolbar_id not in self._APPLICATION_TOOLBARS:
+            raise SpyderAPIError(
+                'Application toolbar "{0}" not found! '
+                'Available toolbars are: {1}'.format(
+                    toolbar_id,
+                    list(self._APPLICATION_TOOLBARS.keys())
+                )
+            )
+
+        return self._APPLICATION_TOOLBARS[toolbar_id]
+
+    def get_application_toolbars(self):
+        """
+        Return all created application toolbars.
+
+        Returns
+        -------
+        list
+            The list of all the added application toolbars.
+        """
+        return self._toolbarslist
+
+    def load_last_visible_toolbars(self):
+        """Loads the last visible toolbars from the options."""
+        # TODO: Move this to a different section since there is no concept
+        # of a MainPlugin (Unless it is created)

```

Comment:
Please solve this in this PR (I already left a comment about it above).

Reply:
Fixed.